components/devtools/actors/tab.rs

```diff
@@ -0,0 +1,123 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+use crate::actor::{Actor, ActorMessageStatus, ActorRegistry};
+use crate::actors::browsing_context::{BrowsingContextActor, BrowsingContextActorMsg};
+use crate::actors::root::RootActor;
+use crate::protocol::JsonPacketStream;
+use msg::constellation_msg::{BrowsingContextId, PipelineId};
+use serde_json::{Map, Value};
+use std::cell::{Cell, RefCell};
+use std::net::TcpStream;
+
+#[derive(Serialize)]
+pub struct TabDescriptorTraits {
+    getFavicon: bool,
+    hasTabInfo: bool,
+    watcher: bool,
+}
+
+#[derive(Serialize)]
+pub struct TabDescriptorActorMsg {
+    actor: String,
+    title: String,
+    url: String,
+    outerWindowID: u32,
+    browsingContextId: u32,
+    traits: TabDescriptorTraits,
+}
+
+#[derive(Serialize)]
+struct GetTargetReply {
+    from: String,
+    frame: BrowsingContextActorMsg,
+}
+
+pub struct TabDescriptorActor {
+    pub name: String,
+    pub title: RefCell<String>,
+    pub url: RefCell<String>,
```

Comment:
Rather than duplicating this data in TabDescriptorActor and BrowsingContextActor, could we choose one of them?